File: xyz.js
// Responses dictionary with added functionalities
const responses = {
    "hello": "Hi there! How can I help you today?",
    "hi": "Hi there! How can I help you today?",
    "about you": "I am Noah, a chatbot. I aim to assist you as best as I can!",
    "how are you": "I'm just a bot, but I'm here to help you!",
    "bye": "Goodbye! Come back soon!",
    "what you can do": "I can tell you the time, solve riddles, share jokes, and more!",
    "joke": "Why don't skeletons fight each other? They don't have the guts!",
    "motivate me": "Believe in yourself! Every day is a new opportunity!",
    "assistant mode": "Assistant mode activated! You can calculate, solve riddles, and more!...to exit assistant mode type :- {exit assistant mode}",
    "exit assistant mode": "Exiting assistant mode. Back to casual chat!",
    "riddle": "I speak without a mouth and hear without ears. I have no body, but I come alive with the wind. What am I?",
    "correct riddle answer": "Correct! The answer is 'Echo'. Well done!",
    "incorrect riddle answer": "Oops, that's not correct. The answer is 'Echo'. Try another challenge!",
    "time": `The current time is ${new Date().toLocaleTimeString()}.`,
};

// Flags and variables for dynamic responses
let inAssistantMode = false;
let awaitingRiddleAnswer = false;

// Add messages to the chat
function addMessage(message, sender) {
    const messagesDiv = document.getElementById("messages");
    const messageDiv = document.createElement("div");
    messageDiv.className = `message ${sender}-message`;

    const icon = document.createElement("img");
    icon.src = sender === "user"
        ? "https://img.freepik.com/premium-vector/user-icon-human-person-symbol-social-profile-icon-avatar-login-sign-web-user-symbol-neumorphic-ui-ux-white-user-interface-web-button-neumorphism-vector-eps-10_399089-2757.jpg?w=1480"
        : "https://i.pinimg.com/736x/0c/67/5a/0c675a8e1061478d2b7b21b330093444.jpg";
    icon.alt = sender;
    icon.className = "message-icon";

    const textDiv = document.createElement("div");
    textDiv.className = "text";
    textDiv.textContent = message;

    if (sender === "bot") {
        messageDiv.appendChild(icon);
        messageDiv.appendChild(textDiv);
        speakMessage(message); // Speak bot responses
    } else {
        messageDiv.appendChild(textDiv);
        messageDiv.appendChild(icon);
    }

    messagesDiv.appendChild(messageDiv);
    messagesDiv.scrollTop = messagesDiv.scrollHeight;
}

// Handle sending messages
function sendMessage() {
    const userInput = document.getElementById("userInput").value.trim().toLowerCase(); // Convert to lowercase for case-insensitive matching
    if (userInput) {
        addMessage(userInput, "user"); // Display user message
        const botResponse = handleResponse(userInput);
        setTimeout(() => addMessage(botResponse, "bot"), 500);
        document.getElementById("userInput").value = ""; // Clear input
    }
}

// Handle dynamic responses
function handleResponse(input) {
    if (awaitingRiddleAnswer) {
        awaitingRiddleAnswer = false;
        if (input === "echo") {
            return responses["correct riddle answer"];
        } else {
            return responses["incorrect riddle answer"];
        }
    }

    if (input === "assistant mode") {
        inAssistantMode = true;
        return responses["assistant mode"];
    }

    if (input === "exit assistant mode") {
        inAssistantMode = false;
        return responses["exit assistant mode"];
    }

    if (inAssistantMode) {
        if (input === "solve riddle") {
            awaitingRiddleAnswer = true;
            return responses["riddle"];
        }
        return "In assistant mode, you can solve riddles or calculate expressions. Try 'solve riddle'.";
    }

    if (input.startsWith("calculate")) {
        let expression = input.replace("calculate", "").trim();
        try {
            let result = eval(expression);
            return `The result of ${expression} is ${result}`;
        } catch (error) {
            return "Oops! I couldn't understand that expression. Try again!";
        }
    }

    if (responses[input]) {
        return responses[input];
    } else {
        return "Sorry, I don't understand that command. Try typing 'joke', 'quiz', or 'calculate'.";
    }
}

// Text-to-Speech (TTS) for bot responses
function speakMessage(message) {
    const utterance = new SpeechSynthesisUtterance(message);
    utterance.lang = "en-US";
    utterance.rate = 1;
    utterance.pitch = 1;
    speechSynthesis.speak(utterance);
}

// Add event listeners
document.getElementById("sendBtn").addEventListener("click", sendMessage);
document.getElementById("userInput").addEventListener("keypress", (e) => {
    if (e.key === "Enter") sendMessage();
});

// Initial bot greeting
window.onload = function () {
    setTimeout(() => addMessage("Hi there! I'm Noah, your voice-enabled chatbot. How can I assist you today?", "bot"), 500);
};

File: xyz.cpp
#include <iostream>
#include <map>
#include <string>
#include <algorithm>
#include <ctime>

using namespace std;

class SimpleChatBot {
private:
    map<string, string> responses;

public:
    SimpleChatBot() {
        responses["hello"] = "Hi there! How can I help you today?";
        responses["hi"] = "Hi there! How can I help you today?";
        responses["about you"] = "I am Noah, a chatbot. Made by Kajal Mantapurwar and Kashish Dhoke. I have limited information, and I hope I can assist you well. For more information, you can ask 'what you can do'.";
        responses["how are you"] = "I'm just a bot, but I'm here to assist you!";
        responses["bye"] = "Goodbye! Come back soon!";
        responses["what you can do"] = "I can tell you the time, tell jokes, give you riddles, and more!";
        responses["tell me a joke"] = "Why don't skeletons fight each other? They don't have the guts!";
        responses["motivate me"] = "Believe in yourself! Every day is a new opportunity!";
        responses["whats the time"] = "Fetching time...";
        responses["tell me time"] = "Fetching time...";
        responses["assistant mode"] = "Assistant mode activated! You can calculate, solve riddles, and more!";
        responses["exit assistant mode"] = "Exiting assistant mode. Back to casual chat!";
    }

    // Get current time
    string getTime() {
        time_t now = time(0);
        tm *ltm = localtime(&now);

        // Format the time as YYYY-MM-DD HH:MM:SS (24-hour format)
        char buffer[80];
        strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S", ltm);
        return string(buffer);
    }

    // Function for solving a riddle
    string solveRiddle() {
        return "I speak without a mouth and hear without ears. I have no body, but I come alive with the wind. What am I?"; // Classic riddle
    }

    // Function to check answer to the riddle
    bool checkRiddleAnswer(string answer) {
        transform(answer.begin(), answer.end(), answer.begin(), ::tolower);  // Make answer case-insensitive

        if (answer == "echo") {
            return true;  // Correct answer
        }
        return false;  // Incorrect answer
    }

    void chat() {
        string user_input;
        bool in_assistant_mode = false;

        cout << "Hello! I'm Noah, your chatbot. How can I assist you today?" << endl;

        while (true) {
            cout << "You: ";
            getline(cin, user_input);

            // Convert to lowercase to make input case-insensitive
            transform(user_input.begin(), user_input.end(), user_input.begin(), ::tolower);

            // Check if in Assistant Mode
            if (in_assistant_mode) {
                if (user_input == "solve riddle") {
                    cout << "Bot: " << solveRiddle() << endl;  // Ask a riddle
                } else if (user_input == "exit assistant mode") {
                    cout << "Bot: " << responses["exit assistant mode"] << endl;
                    in_assistant_mode = false;
                } else {
                    cout << "Bot: Sorry, I didn't understand that. Try again!" << endl;
                }
            } else {
                if (responses.find(user_input) != responses.end()) {
                    if (user_input == "whats the time" || user_input == "tell me time") {
                        cout << "Bot: " << getTime() << endl;
                        cout << "Bot: By the way, did you know? You can activate 'Assistant Mode' for fun challenges and riddles! Just type 'assistant mode' to get started." << endl;
                    } else if (user_input == "hello" || user_input == "hi") {
                        cout << "Bot: " << responses[user_input] << endl;
                    } else if (user_input == "bye") {
                        cout << "Bot: Goodbye! Come back soon!" << endl;
                        break;
                    } else if (user_input == "assistant mode") {
                        cout << "Bot: " << responses[user_input] << endl;
                        in_assistant_mode = true;
                    } else {
                        cout << "Bot: " << responses[user_input] << endl;
                    }
                } else {
                    cout << "Bot: Hmm, I don't know how to respond to that. Could you try asking something else?" << endl;
                }
            }
        }
    }
};

int main() {
    SimpleChatBot bot;
    bot.chat();  // Start the chatbot
    return 0;    // Correct return statement at the end of main function
}
